### PR TITLE
Feature/load connect data to redshift

### PR DESCRIPTION
**Pull request by RiteshAppirio**
Hi @ajefts 
1.  I have added code to load Connect Project and Connect Project Members in a separate class
2. Amazon DMS was failing to load JSON beyond 255 characters despite several efforts.

Questions :
1. We are getting error, during dev load,  for Log Id as it is commented out in [here](https://github.com/topcoder-platform/tcs-loader/blob/dev/src/main/java/com/topcoder/utilities/dwload/tcsredshift/TCLoadTCSRedshiftPost.java#L67). I believe this is nullable field in production.

Notes for changes
1 . Added TCLoadConnectProjects.java to load Connect Postgres Data (Project and Project Members) to Redshift
2.  Added Configuration file run the load.
3.  Added Schema [here](https://github.com/topcoder-platform/tcs-loader/blob/feature/load_connect_data_to_redshift/redshift_schema/06_connect_redshift_schema.sql)

cc @fnisen @vikasrohit

### CODE REVIEW

**Comment by fnisen** [comment]
Thanks a lot @RiteshAppirio. Will be glad to start using this. Had to do a lot of manual clean-up just earlier this week.

**Comment by ajefts** [comment]
@RiteshAppirio For log_id, it is setup in prod as an identity column so it auto increments when you insert new records.  You may need to update the table in the dev env.  Here is the DDL from prod:

'''CREATE TABLE update_log
(
   log_id         integer       DEFAULT "identity"(220606, 0, '1,1'::text) NOT NULL,
   calendar_id    numeric(10),
   log_timestamp  timestamp,
   log_type_id    numeric(3)
);

GRANT INSERT, SELECT, DELETE, UNKNOWN, REFERENCES, UPDATE ON update_log TO tcs_dw;

COMMIT;

'''

File: redshift_schema/06_connect_redshift_schema.sql [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/****************************************************
```
Comment (ajefts) [inline redshift_schema/06_connect_redshift_schema.sql]: @RiteshAppirio Can you add sort and dist keys to these?  We didn't do that for the existing schema (we'll need to add them) and we're suffering on performance. 

For projects, I think the key columns that are filtered/joined on are createdAt, status, type and maybe billingAccountID.

File: scripts/loadconnectdata.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    <load>
+        <classList>
+            <classname>com.topcoder.utilities.dwload.tcspostgresredshift.TCLoadConnectProjects</classname>
+        </classList>
+        <parameterList>
+        </parameterList>
+    </load>
+    <postload>
+        <classname>com.topcoder.utilities.dwload.tcsredshift.TCLoadTCSRedshiftPost</classname>
+        <parameter name="tcs_log_type" value="21"/>
```
Comment (ajefts) [inline scripts/loadconnectdata.xml]: Let's make sure we add this new value to https://raw.githubusercontent.com/topcoder-platform/tcs-loader/feature/load_connect_data_to_redshift/redshift_schema/04_tcs_dw_redshift_data_setup.sql

File: src/main/java/com/topcoder/utilities/dwload/tcspostgresredshift/TCLoadConnectProjects.java [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                    insert.executeUpdate();
+                }
+
+            }
+            log.info("loaded " + count + " records in " + (System.currentTimeMillis() - start) / 1000 + " seconds");
+
+
+        } catch (SQLException sqle) {
+            DBMS.printSqlException(true, sqle);
+            throw new Exception("Load of 'project' table failed.\n" +
```
Comment (ajefts) [inline src/main/java/com/topcoder/utilities/dwload/tcspostgresredshift/TCLoadConnectProjects.java]: should be "projects" table

**Comment by ajefts** [comment]
@RiteshAppirio Do you have a dev env setup now that you're using to test this?

**Comment by RiteshAppirio** [comment]
@ajefts thanks for the review. I am incorporating the changes. I will review the changes with Vikas and checkin on Monday. 
Re: dev environment. yes we have setup Redshift Cluster. However, its not 100% replica of production, as tweaked some of the code and schema to be load the Informix data. We will be working next week to run the TCS Loader from latest git code.

**Comment by ajefts** [comment]
ok, great.  You'll probably need to do an initial bulk load of the data for
the dev env into redshift.  I think the loaders will not support doing a
full load since it'll be too much data for it to process.

For prod, I did exports into json files for the Informix DW tables and then
used Redshift COPY to load it into Redshift.  Then I set the update_log
table to something appropriate based on the initial data load and started
running the loaders...

File: redshift_schema/06_connect_redshift_schema.sql (hunk 1 above)
Comment (RiteshAppirio) [inline redshift_schema/06_connect_redshift_schema.sql]: Composite Sort Key Added on Projects (createdAt, status, "type", billingAccountId);
Dist key is "type"
Sort Key Added on Project Members (createdAt)
dist key = "role"

File: scripts/loadconnectdata.xml (hunk 2 above)
Comment (RiteshAppirio) [inline scripts/loadconnectdata.xml]: Added in Schema
INSERT INTO log_type_lu (log_type_id, log_type_desc) VALUES (21, 'Connect Projects');

File: src/main/java/com/topcoder/utilities/dwload/tcspostgresredshift/TCLoadConnectProjects.java (hunk 3 above)
Comment (RiteshAppirio) [inline src/main/java/com/topcoder/utilities/dwload/tcspostgresredshift/TCLoadConnectProjects.java]: Done Changes

**Comment by RiteshAppirio** [comment]
Hi @ajefts 
I have incorporated the changes. Please check.
Currently doing a full load every time for connect projects, because of an [Issue](https://github.com/topcoder-platform/looker/issues/26)

**Pull request merged**